src/app/signup/components/signup/signup.component.ts

```diff
@@ -13,22 +13,28 @@ import { OAuthManager } from '../../../common/services/oauth-manager.service';
 import { HttpErrorResponse } from '@angular/common/http';
 import { typedFormGroup } from '../../../common/util/typed-forms';
 import { BadgrApiFailure } from '../../../common/services/api-failure';
+import { CaptchaService } from '../../../common/services/captcha.service';
+import { TranslateService } from '@ngx-translate/core';
+import 'altcha';
 
 @Component({
 	selector: 'sign-up',
 	templateUrl: './signup.component.html',
 })
-export class SignupComponent extends BaseRoutableComponent implements OnInit {
+export class SignupComponent extends BaseRoutableComponent implements OnInit, AfterViewInit {
+	baseUrl: string;
 	signupForm = typedFormGroup()
 		.addControl('username', '', [Validators.required, EmailValidator.validEmail])
 		.addControl('firstName', '', Validators.required)
 		.addControl('lastName', '', Validators.required)
 		.addControl('password', '', [Validators.required, Validators.minLength(8)])
 		.addControl('passwordConfirm', '', [Validators.required, this.passwordsMatch.bind(this)])
 		.addControl('agreedTermsService', false, Validators.requiredTrue)
-		.addControl('marketingOptIn', false);
+		.addControl('marketingOptIn', false)
+		.addControl('captcha', '');
 
 	signupFinished: Promise<unknown>;
+	verified = false;
```

Comment:
A more descriptive variable name, indicating *what's* verified would be nice

Reply:
changed to ```captchaVerified```